.github/workflows/dev-container.yml

```diff
@@ -0,0 +1,23 @@
+---
+name: Check research template dev container functions
+
+on:
+  schedule:
+    - cron: "18 8 * * *"
+  workflow_dispatch:
+
+jobs:
+  dev-container:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Checkout research-template-docker repository in subdirectory
+        uses: actions/checkout@v4
+        with:
+          path: 'research-template/research-template-docker'
+  
+      - name: Build and run dev container task
+        uses: devcontainers/ci@a56d055efecd725e8cfe370543b6071b79989cc8 # v0.3.1900000349
+        with:
+          runCmd: ./research-template/research-template-docker/tests/dev_container.sh
```

Comment:
thought: I separated out the package checks from running the OpenSAFELY research-template project and checking RStudio runs.

This is in line with the [acceptance criteria for the associated issue](https://github.com/opensafely-core/codespaces-initiative/issues/18#issue-2159137769).

We could also run this test each time a new image that's being built, but maybe we think this is more testing the other repository. That is, while it's useful to know about failures to fix them, we don't necessarily want to have the image build fail because the research-template is broken.